components/feature/push/src/main/java/mozilla/components/feature/push/MessageBus.kt

```diff
@@ -0,0 +1,76 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ *  License, v. 2.0. If a copy of the MPL was not distributed with this
+ *  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package mozilla.components.feature.push
+
+import androidx.lifecycle.LifecycleOwner
+import mozilla.components.concept.push.Bus
+import mozilla.components.support.base.observer.ObserverRegistry
+
+typealias Registry<T, M> = ObserverRegistry<Bus.Observer<T, M>>
+
+/**
+ * An implementation of [Bus] where the event type is restricted to an enum.
+ */
+class MessageBus<T : Enum<T>, M> : Bus<T, M> {
```

Comment:
The bus was supposed to live in the `support-base` to be multi-purpose. We wanted to refrain from that until we find a better place for it so that `support-base` doesn't become a dumping ground for utility functions. Sebastian recommended we keep it in the feature for now until we know where to move it.

Removing this would be a bigger change. Would doing this in a follow-up work?

Reply:
Sure. Try not to loose track of this.

Reply:
Filed https://github.com/mozilla-mobile/android-components/issues/3171